core/os/os.cpp

```diff
@@ -359,7 +362,46 @@ String OS::get_unique_id() const {
 }
 
 int OS::get_processor_count() const {
+#if defined(_WIN32) || defined(_WIN64)
+	DWORD length = 0;
+	int concurrency = 0;
+	const auto validConcurrency = [&concurrency]() noexcept -> int {
+		if (concurrency == 0)
+			return std::thread::hardware_concurrency();
+		else
+			return concurrency;
+	};
+	if (GetLogicalProcessorInformationEx(RelationAll, nullptr, &length) != FALSE) {
+		return validConcurrency();
+	}
+	if (GetLastError() != ERROR_INSUFFICIENT_BUFFER) {
+		return validConcurrency();
+	}
+	std::unique_ptr<void, void (*)(void *)> buffer(std::malloc(length), std::free);
```

Comment:
Same for `std::unique_ptr`, use `memalloc` / `memfree` instead of custom `std::malloc`. Or `Vector<uint8_t>`.